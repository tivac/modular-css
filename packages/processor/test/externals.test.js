"use strict";

const dedent = require("dedent");
const namer  = require("@modular-css/test-utils/namer.js");
const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("externals", () => {
        let processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });
        
        it("should fail if not a valid composition reference", async () => {
            await expect(processor.string(
                "./invalid-external.css",
                dedent(`
                    :external(some garbage here) { }
                `)
            )).rejects.toThrow(`SyntaxError: Expected`);
        });

        it("should fail on bad class references", async () => {
            const invalid = require.resolve("./specimens/externals-invalid.css");
            
            await expect(processor.file(invalid)).rejects.toThrow(
                `Invalid external reference: nopenopenope`
            );
        });
        
        it("should support overriding external values", async () => {
            await processor.file(
                "./packages/processor/test/specimens/externals.css"
            );

            const { css } = await processor.output();
            
            expect(css).toMatchSnapshot();
        });
    });
});
