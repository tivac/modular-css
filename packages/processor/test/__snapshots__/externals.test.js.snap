// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js externals should support overriding external values 1`] = `
"/* packages/processor/test/specimens/folder/folder.css */
.mc_folder { margin: 2px; }
/* packages/processor/test/specimens/simple.css */
.mc_wooga { color: red; }
/* packages/processor/test/specimens/local.css */
.mc_booga { background: green; }
/* packages/processor/test/specimens/externals.css */
.mc_a .mc_wooga {
    color: green;
}
.mc_b:hover .mc_booga {
    background: blue;
}
.mc_wooga .mc_booga {
    color: red;
}
"
`;
